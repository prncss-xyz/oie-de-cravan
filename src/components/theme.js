const colors = {
  background: '#000000',
  accent: '#ED1F24',
  highLight: '#D8CECC',
  primary: '#929292',
};
const theme = {
  breakpoints: ['785px'],
  colors,
  //fontSizes: [14, 16, 18, 20, 22, 28, 38, 44, 58],
  styles: {
    h1: {
      fontFamily: 'Spectral',
      fontSize: '58px',
      letterSpacing: 1.18,
      lineHeight: '68px',
      textTransform: 'uppercase',
      fontWeight: 'normal',
    },
    authors: {
      fontFamily: 'Spectral',
      fontSize: '38px',
      letterSpacing: 1.18,
      lineHeight: '44px',
      fontWeight: 'normal',
    },
    h2: {
      fontFamily: 'Spectral',
      fontSize: '38px',
      letterSpacing: 1.18,
      lineHeight: '44px',
      fontWeight: 'normal',
      textTransform: 'uppercase',
    },
    navigationMain: {
      fontFamily: 'Spectral',
      fontSize: '19px',
      letterSpacing: 1.18,
      lineHeight: '22px',
    },
    oie: {
      fontFamily: 'Spectral',
      fontSize: '33px',
      lineHeight: '32px',
      fontWeight: 'normal',
    },
    overlay: {
      fontFamily: 'Spectral',
      fontSize: '28px',
      lineHeight: '32px',
      fontWeight: 'normal',
      textTransform: 'uppercase',
    },
    h3: {
      fontFamily: 'Spectral',
      fontSize: '28px',
      lineHeight: '32px',
      fontWeight: 'normal',
    },
    subtitle: {
      fontFamily: 'Arimo',
      fontSize: '20px',
      lineHeight: '26px',
    },
    subtitleFooter: {
      fontFamily: 'Arimo',
      fontSize: '16px',
      lineHeight: '20px',
    },
    body1: {
      fontFamily: 'Spectral',
      fontSize: '22px',
      lineHeight: '32px',
      css: {
        '& p': {
          marginBottom: '32px',
        },
        '& p:last-child': {
          marginBottom: 0,
        },
        '& code': {
          color: colors.accent,
          fontFamily: 'Spectral',
          fontSize: '22px',
          lineHeight: '32px',
        },
        '& h2': {
          // color: colors.accent,
          fontFamily: 'Spectral',
          fontSize: '22px',
          lineHeight: '32px',
          fontWeight: 'bold',
          // marginBottom: '16px',
        },
      },
    },
    body2: {
      fontFamily: 'Spectral',
      fontSize: '18px',
      lineHeight: '24px',
      css: {
        '& p': {
          marginBottom: '24px',
        },
        '& p:last-child': {
          marginBottom: 0,
        },
      },
    },
    caption: {
      fontFamily: 'Spectral',
      fontSize: '16px',
      lineHeight: '20px',
    },
    quote: {
      fontFamily: 'Spectral',
      fontSize: '38px',
      letterSpacing: 1.18,
      fontStyle: 'italic',
      lineHeight: '44px',
    },
    quoteSmall: {
      fontFamily: 'Spectral',
      fontStyle: 'italic',
      fontSize: '28px',
      lineHeight: '32px',
      css: {
        '& p': {
          marginBottom: '32p',
        },
        '& p:last-child': {
          marginBottom: 0,
        },
        '& em': {
          fontStyle: 'normal',
        },

        '& a': {
          fontWeight: 'bold',
          fontStyle: 'normal',
          // textTransform: 'uppercase',
          // textDecoration: 'underline',
        },
      },
    },
    quoteXSmall: {
      fontFamily: 'Spectral',
      fontStyle: 'italic',
      fontSize: '18px',
      lineHeight: '24px',
    },
    button: {
      fontFamily: 'Arimo',
      fontWeight: 'bold',
      fontSize: '16px',
      letterSpacing: 1,
      lineHeight: '20px',
      textTransform: 'uppercase',
    },
    buttonSmall: {
      color: 'accent',
      fontFamily: 'Arimo',
      fontWeight: 'bold',
      fontSize: '14px',
      letterSpacing: 0.75,
      lineHeight: '16px',
      textTransform: 'uppercase',
    },
    navigation: {
      fontFamily: 'Arimo',
      // fontWeight: 'bold',
      fontSize: '16px',
      letterSpacing: 1,
      lineHeight: '20px',
      textTransform: 'uppercase',
    },
    search: {
      fontFamily: 'Arimo',
      fontSize: '16px',
      letterSpacing: 1,
      lineHeight: '20px',
    },
    searchCritaria: {
      fontFamily: 'Arimo',
      fontSize: '14px',
      letterSpacing: 0.75,
      // letterSpacing: 1,
      lineHeight: '20px',
    },
    navigationFooter: {
      fontFamily: 'Arimo',
      fontSize: '14px',
      letterSpacing: 0.75,
      lineHeight: '20px',
      textTransform: 'uppercase',
    },
  },
};

export default theme;
