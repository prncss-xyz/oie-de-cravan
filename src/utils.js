import slugify from '@sindresorhus/slugify';
import { normalize as normalize_ } from '../util';
import { typo_ajust as typo_ajust_ } from '../util';

export const normalize = normalize_;

export const typo_ajust = typo_ajust_;

export const cleanBook = (lang, node) => {
  const linkAuteur = node.data['Auteur'] ?? node.data['Auteur_livre'] ?? '';
  const titre = node.data['Titre'] ?? '';
  return {
    id: node.id,
    auteur: rectAuteur(node.data['Auteur']) ?? '',
    auteurLivre:
      rectAuteur(node.data['Auteur_livre'] ?? node.data['Auteur']) ?? '',
    titre: titre ?? '',
    date: new Date(node.data['Publication__date_']),
    annee: node.data['Publication__date_']?.slice(0, 4),
    collection: node.data['Collection'],
    genre: node.data['Genre'],
    hauteur: node.data['Hauteur__cm_'],
    largeur: node.data['Largeur__cm_'],
    isbn: node.data['ISBN'],
    prixCAD: node.data['Prix_site_web__CAD_'],
    prixEuro: node.data['Prix_site_web__EUR_'],
    pages: node.data['Pages__nombre_'],
    presentation:
      (lang === 'fr' && node.data[`Pr_sentation_et_bio__fr_`]?.trim()) ??
      (lang === 'en' && node.data[`Pr_sentation_et_bio__en_`]?.trim()) ??
      '',
    createursSecondaires:
      (lang === 'fr' && node.data[`Createurs_secondaires_fr`]?.trim()) ??
      (lang === 'en' && node.data[`Createurs_secondaires_en`]?.trim()) ??
      '',
    epuise: !!node.data['_puis_'],
    couvertures: node.data['Couverture']?.map(({ url }) => url) ?? [],
    page: `${lang === 'fr' ? '/livres/' : '/en/books/'}${slugify(
      linkAuteur,
    )}/${slugify(titre)}`,
  };
};

const unPre = /^<p>(.*)<\/p>$/;

/**
 * remove surrounding <p> </p> from html string
 * @param {any} html html string
 */

export const unP = (html) => html.match(unPre)?.[1];

export const rectAuteur = (name) => {
  if (!name) return null;
  let m;
  m = name.match(/(.*),(.*)\((.*)\)/);
  if (m) {
    const [_, a, b, c] = m;
    return `${b.trim()} ${a.trim()} (${c.trim()})`;
  }
  m = name.match(/(.*),(.*)/);
  if (m) {
    const [_, a, b] = m;
    return `${b.trim()} ${a.trim()}`;
  }
  return name.trim();
};
