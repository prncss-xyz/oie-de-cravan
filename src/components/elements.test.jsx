/**
 * @jest-environment jsdom
 */
import { ThemeProvider } from '@emotion/react';
import React from 'react';
import renderer from 'react-test-renderer';
import {unP} from '../utils'

import theme from './theme';
import {
  Arrows,
  Box,
  Flex,
  Grid,
  BookCard,
  Navigation,
  Tilde,
  H1Tilde,
  NavigationFooter,
  Video,
  H2,
  H3,
  GridMd,
  H3Icon,
  VSpacerLarge,
  VSpacerMedium,
  VSpacerSmall,
  VSpacerXSmall,
  Disk,
  Card,
  TextCardMd,
  TextCard,
} from './elements';

describe('Video', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Video />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Box', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Box>contenu</Box>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Flex', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Flex>contenu</Flex>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Grid', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Grid>contenu</Grid>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('BookCard', () => {
  const book = {
    page: 200,
    titre: 'Le titre',
    auteurLivre: 'PrÃ©nom Nom',
    couverture: 'couverture.jpg',
  };
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <BookCard book={book} color='blue' />
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Navigation', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Navigation color='blue'>content</Navigation>
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Tilde', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Tilde />
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<GridMd color='blue' />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('H1Tilde', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <H1Tilde dangerouslySetInnerHTML={{__html: unP('contents')}} color='blue' />
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('H2', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <H2>content</H2>
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('H3', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <H3>content</H3>
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('H3Icon', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <H3Icon
            icon='icon.png'
            dangerouslySetInnerHTML={{__html: unP('<p>contents</p>')}}
            color='blue'
          />
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('VSpacerLarge', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<VSpacerLarge />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('VSpacerMedium', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<VSpacerMedium />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('VSpacerSmall', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<VSpacerSmall />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('VSpacerXSmall', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<VSpacerXSmall />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Arrows', () => {
  it('renders correctly', () => {
    const treeActive = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Arrows dangerouslySetInnerHTML={{__html: unP('<p>contents</p>')}} />
        </ThemeProvider>,
      )
      .toJSON();
    expect(treeActive).toMatchSnapshot();
  });
});

describe('Disk', () => {
  it('renders correctly', () => {
    const treeActive = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Disk active={true} color='blue' />
        </ThemeProvider>,
      )
      .toJSON();
    expect(treeActive).toMatchSnapshot();
    const treeInactive = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Disk active={true} color='blue' />
        </ThemeProvider>,
      )
      .toJSON();
    expect(treeInactive).toMatchSnapshot();
  });
});

// TODO: test Search

describe('Card', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Card pix='20px' piy='30px' color='blue'>contents</Card>
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('TextCardMd', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <TextCardMd color='blue'>contents</TextCardMd>
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});


describe('TextCard', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <TextCard color='blue'>contents</TextCard>
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// TODO: test Clickable
// TODO: should we keep components that do nothing but adding a style? test of so
