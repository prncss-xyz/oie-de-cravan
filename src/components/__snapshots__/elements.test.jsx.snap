// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  color: blue;
  grid-template-columns: repeat(12, 1fr);
}

@media screen and (min-width: 52em) {
  .emotion-0 {
    display: grid;
  }
}

<div
  className="emotion-0 emotion-1"
  color="blue"
  display={
    Array [
      "",
      "",
      "grid",
    ]
  }
/>
`;

exports[`Arrows renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  color: #ED1F24;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-4 {
  box-sizing: border-box;
  min-width: 0;
  padding-left: 18px;
  padding-right: 18px;
  padding-top: 16px;
  padding-bottom: 16px;
  text-align: center;
  font-family: Arimo;
  font-weight: bold;
  font-size: 16px;
  letter-spacing: 1px;
  line-height: 20px;
  text-transform: uppercase;
}

<div
  className="emotion-0 emotion-1"
  color="accent"
>
  <img
    alt="arrow bottom"
    height="24px"
    src="test-file-stub"
  />
  <div
    className="emotion-2 emotion-1"
  >
    <img
      alt="arrow right"
      src="test-file-stub"
      width="22px"
    />
    <div
      className="emotion-4 emotion-5"
      dangerouslySetInnerHTML={
        Object {
          "__html": "contents",
        }
      }
      fontFamily="Arimo"
      fontSize="16px"
      fontWeight="bold"
      letterSpacing={1}
    />
    <img
      alt="arrow left"
      src="test-file-stub"
      width="22px"
    />
  </div>
  <img
    alt="arrow top"
    height="24px"
    src="test-file-stub"
  />
</div>
`;

exports[`BookCard renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  border-style: solid;
  border-color: #929292;
  border-width: 1px;
  padding: 8px;
  background-color: #000000;
}

.emotion-2 {
  box-sizing: border-box;
  min-width: 0;
  border-style: solid;
  border-color: #929292;
  border-width: 1px;
  padding-left: 20px;
  padding-right: 20px;
  padding-top: 20px;
  padding-bottom: 20px;
  width: 100%;
  height: 100%;
}

.emotion-4 {
  box-sizing: border-box;
  min-width: 0;
  padding-bottom: 5px;
  color: #ED1F24;
  font-family: Spectral;
  font-size: 28px;
  line-height: 32px;
  font-weight: normal;
}

.emotion-6 {
  box-sizing: border-box;
  min-width: 0;
  padding-bottom: 20px;
  font-family: Arimo;
  font-size: 20px;
  line-height: 26px;
}

.emotion-8 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<a
  color="blue"
  href={200}
>
  <div
    className="emotion-0 emotion-1"
  >
    <div
      className="emotion-2 emotion-1"
      height="100%"
      width="100%"
    >
      <h3
        className="emotion-4 emotion-1"
        color="accent"
        fontFamily="Spectral"
        fontSize="28px"
        fontWeight="normal"
      >
        Le titre
      </h3>
      <div
        className="emotion-6 emotion-1"
        fontFamily="Arimo"
        fontSize="20px"
      >
        Pr√©nom Nom
      </div>
      <div
        className="emotion-8 emotion-9"
      />
    </div>
  </div>
</a>
`;

exports[`Box renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
}

<div
  className="emotion-0 emotion-1"
>
  contenu
</div>
`;

exports[`Card renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  border-style: solid;
  border-color: #929292;
  border-width: 1px;
  padding: 8px;
  background-color: #000000;
  color: blue;
}

.emotion-2 {
  box-sizing: border-box;
  min-width: 0;
  border-style: solid;
  border-color: #929292;
  border-width: 1px;
  padding-left: 20px;
  padding-right: 20px;
  padding-top: 30px;
  padding-bottom: 30px;
  width: 100%;
  height: 100%;
}

<div
  className="emotion-0 emotion-1"
  color="blue"
>
  <div
    className="emotion-2 emotion-1"
    height="100%"
    width="100%"
  >
    contents
  </div>
</div>
`;

exports[`Disk renders correctly 1`] = `
.emotion-0 {
  height: 10px;
  padding-right: 10px;
}

.emotion-0:last-child {
  padding-right: 0px;
}

<svg
  className="emotion-0"
  color="blue"
  fill="#ED1F24"
  viewBox="0 0 10 10"
>
  <circle
    cx="5"
    cy="5"
    r="5"
  />
</svg>
`;

exports[`Disk renders correctly 2`] = `
.emotion-0 {
  height: 10px;
  padding-right: 10px;
}

.emotion-0:last-child {
  padding-right: 0px;
}

<svg
  className="emotion-0"
  color="blue"
  fill="#ED1F24"
  viewBox="0 0 10 10"
>
  <circle
    cx="5"
    cy="5"
    r="5"
  />
</svg>
`;

exports[`Flex renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="emotion-0 emotion-1"
>
  contenu
</div>
`;

exports[`Grid renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  display: grid;
  grid-template-columns: repeat(12, 1fr);
}

<div
  className="emotion-0 emotion-1"
>
  contenu
</div>
`;

exports[`H1Tilde renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  color: blue;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  box-sizing: border-box;
  min-width: 0;
  max-width: 800px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-family: Spectral;
  font-size: 38px;
  letter-spacing: 1.18px;
  line-height: 44px;
  font-weight: normal;
  text-transform: uppercase;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width: 785px) {
  .emotion-2 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width: 785px) {
  .emotion-2 {
    font-family: Spectral;
    font-size: 58px;
    letter-spacing: 1.18px;
    line-height: 68px;
    font-weight: normal;
  }
}

@media screen and (min-width: 785px) {
  .emotion-2 {
    text-transform: uppercase;
  }
}

.emotion-4 {
  box-sizing: border-box;
  min-width: 0;
}

.emotion-6 {
  box-sizing: border-box;
  min-width: 0;
  padding-left: 18px;
  padding-right: 18px;
  text-align: center;
}

<div
  className="emotion-0 emotion-1"
  color="blue"
>
  <div
    className="emotion-2 emotion-1"
    fontFamily={
      Array [
        "Spectral",
        "Spectral",
      ]
    }
    fontSize={
      Array [
        "38px",
        "58px",
      ]
    }
    fontWeight={
      Array [
        "normal",
        "normal",
      ]
    }
    letterSpacing={
      Array [
        1.18,
        1.18,
      ]
    }
  >
    <div
      className="emotion-4 emotion-5"
    >
      ~
    </div>
    <div
      className="emotion-6 emotion-5"
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
    />
    <div
      className="emotion-4 emotion-5"
    >
      ~
    </div>
  </div>
</div>
`;

exports[`H2 renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  font-family: Spectral;
  font-size: 38px;
  letter-spacing: 1.18px;
  line-height: 44px;
  font-weight: normal;
  text-transform: uppercase;
}

<h2
  className="emotion-0 emotion-1"
  fontFamily="Spectral"
  fontSize="38px"
  fontWeight="normal"
  letterSpacing={1.18}
>
  content
</h2>
`;

exports[`H3 renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  font-family: Spectral;
  font-size: 28px;
  line-height: 32px;
  font-weight: normal;
}

<h3
  className="emotion-0 emotion-1"
  fontFamily="Spectral"
  fontSize="28px"
  fontWeight="normal"
>
  content
</h3>
`;

exports[`H3Icon renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  color: blue;
  text-align: left;
}

.emotion-2 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-4 {
  box-sizing: border-box;
  min-width: 0;
  padding-right: 16px;
}

.emotion-6 {
  box-sizing: border-box;
  min-width: 0;
  min-width: 100%;
  font-family: Spectral;
  font-size: 28px;
  line-height: 32px;
  font-weight: normal;
}

<div
  className="emotion-0 emotion-1"
  color="blue"
>
  <div
    className="emotion-2 emotion-3"
  >
    <div
      className="emotion-4 emotion-1"
    />
    <h3
      className="emotion-6 emotion-1"
      dangerouslySetInnerHTML={
        Object {
          "__html": "contents",
        }
      }
      fontFamily="Spectral"
      fontSize="28px"
      fontWeight="normal"
    />
    <div
      className="emotion-4 emotion-1"
    />
  </div>
</div>
`;

exports[`Navigation renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  font-family: Arimo;
  font-size: 16px;
  letter-spacing: 1px;
  line-height: 20px;
  text-transform: uppercase;
}

<div
  className="emotion-0 emotion-1"
  fontFamily="Arimo"
  fontSize="16px"
  letterSpacing={1}
>
  content
</div>
`;

exports[`TextCard renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  border-style: solid;
  border-color: #929292;
  border-width: 1px;
  padding: 8px;
  background-color: #000000;
  color: blue;
}

.emotion-2 {
  box-sizing: border-box;
  min-width: 0;
  border-style: solid;
  border-color: #929292;
  border-width: 1px;
  padding-left: 30px;
  padding-right: 30px;
  padding-top: 25px;
  padding-bottom: 25px;
  width: 100%;
  height: 100%;
}

<div
  className="emotion-0 emotion-1"
  color="blue"
>
  <div
    className="emotion-2 emotion-1"
    height="100%"
    width="100%"
  >
    contents
  </div>
</div>
`;

exports[`TextCardMd renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  border-color: #929292;
  margin: auto;
  max-width: 682px;
  background-color: #000000;
  color: blue;
}

@media screen and (min-width: 785px) {
  .emotion-0 {
    border-style: solid;
    border-width: 1px;
  }
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    padding: 8px;
  }
}

.emotion-2 {
  box-sizing: border-box;
  min-width: 0;
  width: 100%;
  height: 100%;
}

@media screen and (min-width: 785px) {
  .emotion-2 {
    border-style: solid;
    border-color: #929292;
    border-width: 1px;
  }
}

@media screen and (min-width: 40em) {
  .emotion-2 {
    padding-left: 30px;
    padding-right: 30px;
    padding-top: 25px;
    padding-bottom: 25px;
  }
}

<div
  className="emotion-0 emotion-1"
  color="blue"
>
  <div
    className="emotion-2 emotion-1"
    height="100%"
    width="100%"
  >
    contents
  </div>
</div>
`;

exports[`Tilde renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  font-family: Spectral;
  font-size: 58px;
  letter-spacing: 1.18px;
  line-height: 68px;
  font-weight: normal;
  text-transform: uppercase;
}

<div
  className="emotion-0 emotion-1"
  fontFamily="Spectral"
  fontSize="58px"
  fontWeight="normal"
  letterSpacing={1.18}
>
  ~
</div>
`;

exports[`VSpacerLarge renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  padding-bottom: 80px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    padding-bottom: 180px;
  }
}

<div
  className="emotion-0 emotion-1"
/>
`;

exports[`VSpacerMedium renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  padding-bottom: 60px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    padding-bottom: 100px;
  }
}

<div
  className="emotion-0 emotion-1"
/>
`;

exports[`VSpacerSmall renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  padding-bottom: 40px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    padding-bottom: 60px;
  }
}

<div
  className="emotion-0 emotion-1"
/>
`;

exports[`VSpacerXSmall renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  padding-bottom: 40px;
}

<div
  className="emotion-0 emotion-1"
/>
`;

exports[`Video renders correctly 1`] = `
<iframe
  allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
  allowFullScreen={true}
  frameBorder="0"
  height="100%"
  mozallowfullscreen="true"
  webkitallowfullscreen="true"
  width="100%"
/>
`;
